seed-job/template.yaml

```diff
@@ -140,6 +141,16 @@ objects:
                             sh "PYTHONPATH=${PIPELINE_REPO_DIR} python ${PIPELINE_REPO_DIR}/ccp/notifications/notify.py failed ${NAMESPACE} `oc get route jenkins -o template --template={{.spec.host}}` ${image_name} ${BUILD_NUMBER} ${PIPELINE_NAME}"
                             }
                     }
+                      if (image_is_built == true) {
```

Comment:
It makes more sense to keep this check as a part of image removal stage rather than outside it. For example for pre-build stage, we do the comparison inside the stage. 

More importantly, we're doing everything in stages (not literally in Jenkinsfile terminology but as a way of doing things.) Doing this outside the stage is looking unintuitive.

Reply:
IMO - not defining a stage which is not needed in pipeline makes more sense.

What do you think @bamachrn @mohammedzee1000 ?

Reply:
It makes sense, sure. But we're doing everything inside stages in the new architecture. It's not a rule that we've set on ourselves and we can remove for example, pre-build, out of the stage. But till we take that call, can we adhere to keeping things uniform?

Reply:
I see conditionals, stage definitions as groovy language representation of Jenkinsfile for pipelines. 
The respective operations per stage is grouped into stage definitions only. The conditionals and variables defined (and stages themselves) are then grouped into the `node` definition.
For example the defaults set in variables, for eg: `success=False` controls the operations inside the stage (for eg: Notify user stage depends on that variable).